### PR TITLE
[BugFix] Fix an issue where the numberConnection variable in the ConnectScheduler class repeatedly counts the same connection

### PR DESCRIPTION
**Pull request by zhangheihei**
## Why I'm doing:
fe_connection_total metrics counts the same connection twice, then will result 'Reach limit of connections' error.
## What I'm doing:
fix the metircs counter

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [x] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

**Review by kangkaisen — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectScheduler.java [hunk 1]
```diff
@@ -139,9 +139,11 @@ public boolean registerConnection(ConnectContext ctx) {
         if (currentConn >= currentMaxConn) {
             return false;
         }
-        numberConnection.incrementAndGet();
-        connCountByUser.get(ctx.getQualifiedUser()).incrementAndGet();
-        connectionMap.put((long) ctx.getConnectionId(), ctx);
+        ConnectContext existingConnectionContext = connectionMap.putIfAbsent((long) ctx.getConnectionId(), ctx);
+        if (existingConnectionContext == null) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/qe/ConnectScheduler.java:143]: Do you know when this will be null?

Comment (zhangheihei) [reply fe/fe-core/src/main/java/com/starrocks/qe/ConnectScheduler.java:143]: When a connection is established,registerConnection has two callers:
1 handleEvent->registerConnection
2 handleEvent->connectScheduler.submit->executor.submit->LoopHandler.run->registerConnection
The first caller to registerConnection enters the existingConnectionContext == null block of code。

**Review by LiShuMing — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Comment by zhangheihei** [comment]
hi @kangkaisen
 How should I pass the ADMIT TEST CI PIPELINE?

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectScheduler.java [hunk 2]
```diff
@@ -139,9 +139,11 @@ public boolean registerConnection(ConnectContext ctx) {
         if (currentConn >= currentMaxConn) {
             return false;
         }
-        numberConnection.incrementAndGet();
-        connCountByUser.get(ctx.getQualifiedUser()).incrementAndGet();
-        connectionMap.put((long) ctx.getConnectionId(), ctx);
+        ConnectContext existingConnectionContext = connectionMap.putIfAbsent((long) ctx.getConnectionId(), ctx);
```
Comment (trueeyu) [inline fe/fe-core/src/main/java/com/starrocks/qe/ConnectScheduler.java:142]: Every time a link is created, it will be registered twice?

**Review by gengjun-git — CHANGES_REQUESTED** [review]
Can you show the bug details? Which version did you encounter this bug in?

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectScheduler.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/qe/ConnectScheduler.java:143]: These two calls will not occur at the same time. One is for NIO, anther is for Block IO.

**Comment by zhangheihei** [comment]
> Can you show the bug details? Which version did you encounter this bug in?

Sorry, I was wrong about the logic of the scenario for that bugfix

**Pull request closed without merging**